rgw/COSBench/cosbench.py

```diff
@@ -0,0 +1,269 @@
+"""Module to trigger COS Bench workloads."""
+import argparse
+import json
+import logging
+import subprocess
+import sys
+from os.path import dirname, join
+from tempfile import NamedTemporaryFile
+from typing import Dict, Optional, Tuple
+
+from jinja2 import Template
+from yaml import safe_load
+
+APP_DIR = "/opt/cosbench"
+LOG = logging.getLogger(__name__)
+
+
+def exec_command(cmd: str) -> subprocess.CompletedProcess:
+    """Execute the given command.
+
+    Args:
+        cmd (str):  The command that needs to be executed.
+
+    Returns:
+
+    """
+    LOG.debug("Executing command: %s", cmd)
+    return subprocess.run([cmd], shell=True, check=True, stdout=subprocess.PIPE)
+
+
+def get_user_info() -> Dict:
+    """Returns cosbench01 user info available in realm."""
+    try:
+        proc = exec_command(cmd="radosgw-admin user info --uid cosbench01")
+    except subprocess.CalledProcessError:
+        proc = exec_command(
+            cmd="radosgw-admin user create --uid cosbench01 "
+            "--display-name cosbench01 --email cosbench01@noreply.com"
+        )
+
+    return json.loads(proc.stdout)
+
+
+def get_swauth_secret() -> str:
+    """Returns the secret key of cosbench01:swift user.
+
+    Creates the user if not found in realm. The sub user required for swift auth is also
+    created.
+
+    Returns:
+        secret_key  of the standard user.
+    """
+    user_info = get_user_info()
+
+    if not user_info["subusers"]:
+        exec_command(
+            cmd="radosgw-admin subuser create --uid cosbench01 "
+            "--subuser cosbench01:swift --access full"
```

Comment:
Same as above. Can we have support for creating custom users apart from default ones?